lib/actions/airtable/airtable.js

```diff
@@ -81,26 +78,190 @@ class AirtableAction extends Hub.Action {
             return new Hub.ActionResponse(response);
         });
     }
-    form() {
+    checkBaseList(request) {
+        return __awaiter(this, void 0, void 0, function* () {
+            if (request.params.state_json) {
+                const stateJson = JSON.parse(request.params.state_json);
+                const response = yield this.refreshTokens(stateJson.tokens.refresh_token);
+                return gaxios.request({
+                    method: "GET",
+                    url: "https://api.airtable.com/v0/meta/bases",
+                    headers: {
+                        Authorization: `Bearer ${response.data.access_token}`,
+                    },
+                }).catch((err) => {
+                    winston.error(JSON.stringify(err));
+                    throw "Error listing bases, oauth probably bad.";
+                });
+            }
+            else {
+                return null;
```

Comment:
Can we add logging here to make it more clear why we did not return null?  I think this would be "json state not found - OAuth missing" right?